plugin/system_copy.vim

```diff
@@ -33,23 +42,55 @@ function! s:resolve_mode(type, arg)
   endif
 endfunction
 
-function! s:CopyCommandForCurrentOS()
+function! s:currentOS()
   let os = substitute(system('uname'), '\n', '', '')
-  if has("gui_mac") || os == 'Darwin'
-    return 'pbcopy'
+  let known_os = 'unknown'
```

Comment:
In my opinion declaring variables inside conditions isn't good practice so I added this line to your (@christoomey) proposal.